static/js/publisher/market/market.js

```diff
@@ -12,6 +12,91 @@ function initSnapIconEdit(iconElId, iconInputId) {
   });
 }
 
+function initSnapScreenshotsEdit(screenshotsToolbarElId, screenshotsWrapperElId, data) {
+  // DOM elements
+  const screenshotsToolbarEl = document.getElementById(screenshotsToolbarElId);
+  const screenshotsWrapper = document.getElementById(screenshotsWrapperElId);
+
+  // simple state handling (and serializing as JSON in hidden input)
+  const state = {};
+  const stateInput = document.createElement('input');
+  stateInput.type = "hidden";
+  stateInput.name = "state";
+
+  screenshotsToolbarEl.parentNode.appendChild(stateInput);
+
+  const setState = function(nextState) {
+    for (let key in nextState) {
+      if (nextState.hasOwnProperty(key)) {
+        state[key] = nextState[key];
+      }
+    }
+
+    stateInput.value = JSON.stringify(state);
+  };
+
+  setState({
+    screenshots: data.map((url) => { return { url }; })
+  });
+
+  // templates
+  const screenshotTpl = (screenshot) => `
+    <div class="col-2">
+      <img src="${screenshot.url}" alt="" />
+    </div>
+  `;
+
+  const emptyTpl = () => `
+    <div class="col-12">
+      <a class="p-empty-add-screenshots js-add-screenshots">Add images</a>
+    </div>
+  `;
+
+  const renderScreenshots = (screenshots) => {
+    if (screenshots.length) {
+      screenshotsWrapper.innerHTML = screenshots.map(screenshotTpl).join("");
+    } else {
+      screenshotsWrapper.innerHTML = emptyTpl();
+    }
+  };
+
+  const render = () => {
+    renderScreenshots(state.screenshots);
+  };
+
+  render();
+
+  const onScreenshotsChange = function() {
+    const fileList = this.files;
+
+    for (var i = 0; i < fileList.length; i++) {
```

Comment:
`for(const file in fileList)`

Reply:
Tried that, but it doesn't work. FileList is not really an array, just and array-like object.